packages/react/tooltip/src/Tooltip.tsx

```diff
@@ -14,45 +13,114 @@ import { createPopperScope } from '@radix-ui/react-popper';
 import { Portal } from '@radix-ui/react-portal';
 import { Slottable } from '@radix-ui/react-slot';
 import * as VisuallyHiddenPrimitive from '@radix-ui/react-visually-hidden';
-import { useCallbackRef } from '@radix-ui/react-use-callback-ref';
 import { useId } from '@radix-ui/react-id';
-import { createStateMachine } from './createStateMachine';
-import { tooltipStateChart } from './tooltipStateChart';
 
 import type * as Radix from '@radix-ui/react-primitive';
 import type { Scope } from '@radix-ui/react-context';
 
 /* -------------------------------------------------------------------------------------------------
- * State machine
+ * TooltipProvider
  * -----------------------------------------------------------------------------------------------*/
 
-type StateAttribute = 'closed' | 'delayed-open' | 'instant-open';
-const stateMachine = createStateMachine(tooltipStateChart);
+const PROVIDER_NAME = 'TooltipProvider';
+const DEFAULT_DELAY_DURATION = 700;
+const TOOLTIP_OPEN = 'tooltip.open';
+
+type ScopedProps<P = {}> = P & { __scopeTooltipProvider?: Scope };
+const [createTooltipContext, createTooltipScope] = createContextScope(PROVIDER_NAME, [
+  createPopperScope,
+]);
+const usePopperScope = createPopperScope();
+
+type TooltipProviderContextValue = {
+  isOpenDelayed: boolean;
+  delayDuration: number;
+  onTooltipOpen(): void;
+  onTooltipClose(): void;
+};
+
+const [TooltipProviderProvider, useTooltipProviderContext] =
+  createTooltipContext<TooltipProviderContextValue>(PROVIDER_NAME, {
+    isOpenDelayed: true,
+    delayDuration: DEFAULT_DELAY_DURATION,
+    onTooltipOpen: () => {},
+    onTooltipClose: () => {},
+  });
```

Comment:
Supplying defaults so that the `TooltipProvider` isn't required. Without it, tooltips will all have a delay before opening.